account-kit/signer/src/apiKeySigner.ts

```diff
@@ -0,0 +1,246 @@
+import { type ConnectionConfig, type SmartAccountSigner } from "@aa-sdk/core";
+import { ApiKeyStamper } from "@turnkey/api-key-stamper";
+import type {
+  AlchemySignerClientEvents,
+  AuthenticatingEventMetadata,
+  CreateAccountParams,
+  EmailAuthParams,
+  EnableMfaParams,
+  EnableMfaResult,
+  GetWebAuthnAttestationResult,
+  MfaFactor,
+  OauthConfig,
+  OauthParams,
+  OtpParams,
+  RemoveMfaParams,
+  SignupResponse,
+  SubmitOtpCodeResponse,
+  User,
+  VerifyMfaParams,
+} from "./client/types.js";
+import { BaseSignerClient } from "./client/base.js";
+import { SmartAccountSignerFromClient } from "./smartAccountSigner.js";
+
+export interface AlchemyApiKeySignerClientParams {
+  connection: ConnectionConfig;
+  apiKey: {
+    publicKey: string;
+    privateKey: string;
+  };
+}
+
+/**
+ * AlchemyApiKeySignerClient is a client for signing messages using an API key.
+ * It extends the BaseSignerClient and uses the ApiKeyStamper for signing.
+ * Primarily intended to be used server-side.
+ */
+export class AlchemyApiKeySignerClient extends BaseSignerClient<undefined> {
```

Comment:
nit: can we put this in `src/client/apiKey` or something and then below have `createApiKeySigner` be the only thing exported from src/apiKeySigner?

Reply:
sure thing